### PR TITLE
Update App Inventor to target Android SDK 29

### LINKED ISSUE #2173: Update App Inventor to SDK 29
New apps compiled with App Inventor need to target SDK 29 by August 1st, 2020 in order to be accepted by Google Play.

Documentation of potential affecting changes here: https://docs.google.com/document/d/1JZH9ZXffsU69V2tYsj1Recd02E2isS0ZLAe3Kv2Mp7g/edit?usp=sharing

### PR DESCRIPTION
**Pull request by ewpatton**
Resolves #2173 

Todo:
- [ ] Handle user-provided paths starting with /sdcard and /storage
- [ ] Integrate jetifier into buildserver to handle extensions that rely on older Android support libraries
- [ ] Update all documentation references to /sdcard

Test project I've been using: [SDK29Test.aia.zip](https://github.com/mit-cml/appinventor-sources/files/4523939/SDK29Test.aia.zip)

Change-Id: I9c2e0b4531e9ff9242e108fe6427acc7a5ab0739

### CODE REVIEW

**Comment by pavi2410** [comment]
May I help you with this?

> Integrate jetifier into buildserver to handle extensions that rely on older Android support libraries

I have worked on Jetifier integration into Kodular. Although it is pending due to a DexException caused by a conflicting definition.

**Comment by ewpatton** [comment]
@pavi2410 Sure. You should be able to submit a PR to my branch.

**Comment by Mika-89** [comment]
@ewpatton 
We still can write and read external storage files with Android 10.
It will be finally only changed at Android 11.

Just add inside of the Manifest file following line inside the application tag:

`android:requestLegacyExternalStorage="true"`

https://developer.android.com/preview/privacy/storage

So I think you should do some changes at your "QUtil" class since it's still possible until Android 11.

**Comment by ewpatton** [comment]
Thanks for the info @Mika-89. I'll think about how we might best address that.

**Comment by jisqyv** [comment]
@ewpatton I just pushed three commits that deal with the Web component permission issues.

File: appinventor/components/src/com/google/appinventor/components/runtime/CloudDB.java [hunk 1]
```diff
@@ -1130,6 +1130,10 @@ private void checkProjectIDNotBlank(){
     }
   }
 
+  public Form getForm() {
```
Comment (jisqyv) [inline appinventor/components/src/com/google/appinventor/components/runtime/CloudDB.java:1133]: @ewpatton Where is this used?

**Comment by jisqyv** [comment]
@ewpatton So fetching images from CloudDB in a compiled app fails (silently!) due to a lack of storage permission. I'm working on a fix.

**Comment by jisqyv** [comment]
@ewpatton P.S. It looks like it is broken in the current release as well.

**Comment by jisqyv** [comment]
@ewpatton The latest commit fixes permission for CloudDB

File: appinventor/components/src/com/google/appinventor/components/runtime/Canvas.java [hunk 2]
```diff
@@ -754,6 +770,23 @@ public Canvas(ComponentContainer container) {
     mGestureDetector = new GestureDetector(context, new FlingGestureListener());
   }
 
+  public void Initialize() {
+    // Note: The code below does not call ourselves after the
+    // onGranted because we don't do anything beyond getting
+    // permissions. If we ever add code to this Initialize method,
+    // then be sure to call ourselves in onGranted()
```
[the remark this answers is not in the document]
Comment (jisqyv) [reply appinventor/components/src/com/google/appinventor/components/runtime/Canvas.java]: Picky picky :-) I've updated the comment.

**Pull request merged**